src/openrpc/lifecycle.json

```diff
@@ -1,261 +1,494 @@
+          },
+          {
+            "name": "capabilities",
+            "x-uses": [
+              "xrn:firebolt:capability:lifecycle:state"
+            ]
+          }
+        ],
+        "summary": "Listen to the suspended event",
+        "params": [],
+        "result": {
+          "name": "value",
+          "schema": {
+            "$ref": "#/components/schemas/LifecycleEvent"
+          }
+        },
+        "examples": [
+          {
+            "name": "Default Example",
+            "params": [],
+            "result": {
+              "name": "Default Result",
+              "value": {
+                "state": "paused",
+                "previous": "suspended"
+              }
+            }
+          }
+        ]
+      },
+      {
+        "name": "onHibernate",
+        "tags": [
+          {
+            "name": "event"
+          },
+          {
+            "name": "capabilities",
+            "x-uses": [
+              "xrn:firebolt:capability:lifecycle:state"
+            ]
+          }
+        ],
+        "summary": "Listen to the hibernating event",
+        "params": [],
+        "result": {
+          "name": "value",
+          "schema": {
+            "$ref": "#/components/schemas/LifecycleEvent"
+          }
+        },
+        "examples": [
+          {
+            "name": "Default Example",
+            "params": [],
+            "result": {
+              "name": "Default Result",
+              "value": {
+                "state": "hibernated",
+                "previous": "suspended"
+              }
+            }
+          }
+        ]
+      },
+      {
+        "name": "onRestore",
+        "tags": [
+          {
+            "name": "event"
+          },
+          {
+            "name": "capabilities",
+            "x-uses": [
+              "xrn:firebolt:capability:lifecycle:state"
+            ]
+          }
+        ],
+        "summary": "Listen to the restoration event",
+        "params": [],
+        "result": {
+          "name": "value",
+          "schema": {
+            "$ref": "#/components/schemas/LifecycleEvent"
+          }
+        },
+        "examples": [
+          {
+            "name": "Default Example",
+            "params": [],
+            "result": {
+              "name": "Default Result",
+              "value": {
+                "state": "suspended",
+                "previous": "hibernated"
+              }
+            }
+          }
+        ]
+      },
+      {
+        "name": "onDestroy",
+        "tags": [
+          {
+            "name": "event"
+          },
+          {
+            "name": "capabilities",
+            "x-uses": [
+              "xrn:firebolt:capability:lifecycle:state"
+            ]
+          }
+        ],
+        "summary": "Listen to the destroy event",
+        "params": [],
+        "result": {
+          "name": "value",
+          "schema": {
+            "$ref": "#/components/schemas/LifecycleEvent"
```

Comment:
I'm wondering if we should use a separate schema instead of the shared `LifecycleEvent` schema for this method's response, as `destroy` isn't really a lifecycle state on its own--this is just a notification that the app will be destroyed.

We will also want to add a `reason` string property to this event so the app can understand why its being destroyed.